src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4356,14 +4361,27 @@ protected override void VisitArguments(ImmutableArray<BoundExpression> arguments
             {
                 if (HasImplicitTypeArguments(node))
                 {
-                    var binder = (node as BoundCall)?.BinderOpt ?? (node as BoundCollectionElementInitializer)?.BinderOpt ?? throw ExceptionUtilities.UnexpectedValue(node);
+                    var binder = node switch
+                    {
+                        BoundCall { BinderOpt: { } b } => b,
+                        BoundCollectionElementInitializer { BinderOpt: { } b } => b,
+                        BoundForEachStatement { EnumeratorInfoOpt: { Binder: var b } } => b,
```

Comment:
>var [](start = 77, length = 3)

This should be `{ }`, or it will match a `null` binder.

Reply:
This is not opt and cannot be null. Should we still check for it?

Reply:
We should try and be equivalent to the logic we're replacing.